Rbigcview

brisbori opened the pull request:
Revises changes made to task drawing routine, adds TP sectors

brisbori committed:
Fix errors where there are C records but no turning points declared (known issue with EW barographs)
Alter display of task information to show start, finish and TP numbers
No longer show track from takeoff to start, or finish to landing in task layer (clearer display when we draw start and finish lines)

brisbori committed:
Created two new variables in the model.task object, "takeoff" and "landing" initialised empty
Moved modification of the IgcFile.task array to the end of the  parseIGC function
Task is now displayed only if the IgcFile.task array is populated, AND the start latitude is not at the equator- new fix for EW issue
Added leaflet plug-in Semicircle.js from Jan Pieter Waagmeester (see jieter/Leaflet-semicircle) to permit sector drawing
Added routines to draw start and finish lines and turning points.

brisbori committed:
Further change to ensure takeoff and landing are only shown if they have been entered in the declaration

alistairmgreen commented on mapcontrol.js:
```diff
@@ -55,13 +55,97 @@ function createMapControl(elementName) {
+}    
+
+
+function getline(pt1,pt2,linerad,drawOptions) {
```
I would be tempted to put these new functions inside `createMapControl()` so that they don't have global scope.

The `createMapControl` function was designed to create a closure to hide the internal workings of the map from the rest of the program.

brisbori replied:
Hi Alistair

Now fixed the bug and moved the functions (except "getBearing()), which 
I reckon is more generic.  See latest commit.

regards

On 02/09/15 21:31, Alistair Green wrote:

> In mapcontrol.js 
> https://github.com/alistairmgreen/jsigc/pull/12#discussion_r38580413:
> 
> > ```
> >          }
> >          mapLayers.task = L.layerGroup(taskLayers).addTo(map);
> >          layersControl.addOverlay(mapLayers.task, 'Task');
> >      }
> >  };
> > ```
> > 
> > +}
> > +
> > +
> > +function getline(pt1,pt2,linerad,drawOptions) {
> 
> I would be tempted to put these new functions inside 
> |createMapControl()| so that they don't have global scope.
> 
> The `createMapControl
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/alistairmgreen/jsigc/pull/12/files#r38580413.

brisbori committed:
Fixed bug where some TP sectors were 180 deg. out.
Moved functions for drawing start and finish lines and tp sectors inside createMapControl()

alistairmgreen commented:
Thanks, this seems to work correctly now.
